src/QirRuntime/test/QIR-static/qsharp/qir-test-other.qs

```diff
@@ -4,10 +4,32 @@
 namespace Microsoft.Quantum.Testing.QIR.Other {
 
     open Microsoft.Quantum.Bitwise;
+    open Microsoft.Quantum.Convert;
+
+    function PauliArrayAsIntTest() : Int {
+        if 0 != PauliArrayAsInt([PauliI]) { return  1; }   // The return value indicates which test case has failed.
+        if 1 != PauliArrayAsInt([PauliX]) { return  2; }
+        if 3 != PauliArrayAsInt([PauliY]) { return  3; }
+        if 2 != PauliArrayAsInt([PauliZ]) { return  4; }
+
+        if 0x2310 != PauliArrayAsInt(
+            [PauliI, PauliI,  PauliX, PauliI,  PauliY, PauliI,  PauliZ, PauliI]) { return  5; }
+
+        // The Pauli array items are default-initialized to PauliI.
+        if 0 != PauliArrayAsInt(new Pauli[31]) { return  6; }
```

Comment:
We should check with @samarsha about QEP 2 timelines, and when `new Pauli[31]` will be deprecated in favor of `[PauliI, size=31]`.

Reply:
Probably no later than the April release; March release is possible too, but depends on the HM type inference being wrapped up quickly.

Reply:
I added a comment in this line 
```q#
// Todo: Replace `new Pauli[31]` with `[PauliI, size=31]` when that is implemented.
```